test/13_arraycommon.jl

```diff
@@ -1827,4 +1827,21 @@ end
     end
 end
 
+@testset "vector of unordered" begin
+    struct UnorderedBar
+        a::String
+    end
+    
+    x0 = [UnorderedBar("s$i") for i in 1:10]
+    x = CategoricalArray(x0)
+    @test x[5] == UnorderedBar("s5")
+    @test x[10] == UnorderedBar("s10")
```

Comment:
I don't see the point of these checks. Just do `x == x0` if you want to check that all values are equal?

Reply:
done